cpp/oneapi/dal/table/homogen.hpp

```diff
@@ -115,6 +115,33 @@ class ONEDAL_EXPORT homogen_table : public table {
         return homogen_table{ data, row_count, column_count, layout };
     }
 
+#ifdef ONEDAL_DATA_PARALLEL
+    /// Creates a new ``homogen_table`` instance from an array.
+    /// The created table shares data ownership with the given array.
+    ///
+    /// @tparam Data        The type of elements in the data block that will be stored into the
+    ///                     table. The table initializes data types of metadata with this data type.
+    ///                     The feature types should be set to default values for :literal:`Data` type:
+    ///                     contiguous for floating-point, ordinal for integer types. The :literal:`Data`
+    ///                     type should be at least :expr:`float`, :expr:`double` or :expr:`std::int32_t`.
+    ///
+    /// @param data         The array that stores a homogeneous data block.
+    /// @param row_count    The number of rows in the table.
+    /// @param column_count The number of columns in the table.
+    /// @param dependencies Events indicating availability of the :literal:`Data` for reading or writing.
+    /// @param layout       The layout of the data. Should be :literal:`data_layout::row_major` or
+    ///                     :literal:`data_layout::column_major`.
+    template <typename Data>
+    static homogen_table wrap(const array<Data>& data,
+                              std::int64_t row_count,
+                              std::int64_t column_count,
+                              const sycl::vector_class<sycl::event>& dependencies,
+                              data_layout layout = data_layout::row_major) {
```

Comment:
1. Is it a proper order of these params?
2. `std::vector` instead of `sycl::vector_class`?

Reply:
I made this function consistent with another wrap
https://github.com/oneapi-src/oneDAL/blob/master/cpp/oneapi/dal/table/homogen.hpp#L80

Reply:
@michael-smirnov @rlnx, Is this wrap suitable for us or do we need changes?

Reply:
I think we need to remove this... The problem is that `dal::array` is already expected to track dependencies as it can be created from pointer and deps vector. We already have constructor in this table that accepts just an array without dependencies. I assume, if the user wants to create table with dependencies, they should create array with dependencies first and then pass to the `wrap` function. 

I suggest to think more about this function and to not introduce it in this PR as it could create a lot of problems with backward compatibility in future.

What I suggest now is to parametrize `ndarray::flatten` with additional dependencies argument, so that:
```c++
const auto array_that_encloses_dependencies = ndarr.flatten(queue, { some_event }); 
homogen_table::wrap(array_that_encloses_dependencies, row_count, column_count);
```